core/templates/dev/head/components/AnswerGroupEditorDirective.js

```diff
@@ -239,13 +225,31 @@ oppia.directive('answerGroupEditor', [function() {
           return $scope.activeRuleIndex !== -1;
         };
 
-        $scope.$on('onInteractionIdChanged', function() {
+        // Updates answer choices when the interaction requires it -- e.g.,
+        // the rules for multiple choice need to refer to the multiple choice
+        // interaction's customization arguments.
+        // TODO(sll): Remove the need for this watcher, or make it less ad hoc.
+        var cleanup1 = $scope.$on('updateAnswerChoices', function() {
+          $scope.answerChoices = $scope.getAnswerChoices();
+        });
+
+        var cleanup2 = $scope.$on('externalSave', function() {
+          if ($scope.isRuleEditorOpen()) {
+            $scope.saveRules();
+          }
+        });
+
+        var cleanup3 = $scope.$on('onInteractionIdChanged', function() {
           if ($scope.isRuleEditorOpen()) {
             $scope.saveRules();
           }
           $scope.$broadcast('updateAnswerGroupInteractionId');
           $scope.answerChoices = $scope.getAnswerChoices();
         });
+
+        $scope.$on('$destroy', cleanup1);
```

Comment:
I'm not keen on variables named cleanup, or the requirement to do this extra step everytime we want to register a listener for a globally broadcasted Angular message. Can we add some sort of utility function shared among the entire Oppia module to pipeline this for us? Maybe something like:

```javascript
BroadcastService.on($scope, 'updateAnswerChoices', function() {
  ...
});
```

Where ``BroadcastService.on`` would look something like:

```javascript
on: function($scope, eventName, callback) {
  var cleanup = $scope.$on(eventName, callback);
  $scope.$on('$destroy', cleanup);
}
```

It would be nicer if we could add a custom function to ``$scope`` itself, but I don't know how feasible that is.